receiver/awscontainerinsightreceiver/internal/gpu/dcgmscraper_config.go

```diff
@@ -0,0 +1,149 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package gpu
+
+import (
+	"context"
+	"time"
+
+	"github.com/prometheus/common/model"
+	"github.com/prometheus/prometheus/config"
+	"github.com/prometheus/prometheus/discovery"
+	"github.com/prometheus/prometheus/discovery/kubernetes"
+	"github.com/prometheus/prometheus/model/relabel"
+	"go.opentelemetry.io/collector/component"
+	"go.opentelemetry.io/collector/consumer"
+	"go.opentelemetry.io/collector/receiver"
+	"go.uber.org/zap"
+
+	ci "github.com/open-telemetry/opentelemetry-collector-contrib/internal/aws/containerinsight"
+	"github.com/open-telemetry/opentelemetry-collector-contrib/receiver/awscontainerinsightreceiver/internal/prometheusscraper"
+)
+
+const (
+	caFile                    = "/var/run/secrets/kubernetes.io/serviceaccount/ca.crt"
+	collectionInterval        = 60 * time.Second
+	jobName                   = "containerInsightsDCGMExporterScraper"
+	scraperMetricsPath        = "/metrics"
+	scraperK8sServiceSelector = "k8s-app=dcgm-exporter-service"
+)
+
+type DcgmScraper struct {
+	ctx                context.Context
+	settings           component.TelemetrySettings
+	host               component.Host
+	hostInfoProvider   hostInfoProvider
+	prometheusReceiver receiver.Metrics
+	k8sDecorator       prometheusscraper.Decorator
+	running            bool
+}
+
+type DcgmScraperOpts struct {
+	Ctx               context.Context
+	TelemetrySettings component.TelemetrySettings
+	Consumer          consumer.Metrics
+	Host              component.Host
+	HostInfoProvider  hostInfoProvider
+	K8sDecorator      prometheusscraper.Decorator
+	Logger            *zap.Logger
+}
+
+type hostInfoProvider interface {
+	GetClusterName() string
+	GetInstanceID() string
+}
+
+func GetScraperConfig(hostInfoProvider hostInfoProvider) *config.ScrapeConfig {
```

Comment:
If you're moving it here, shouldn't it be removed from internal/gpu/dcgmscraper.go?

Reply:
Yes will delete it correct, again mad merge